OpenEdXMobile/src/main/java/org/edx/mobile/view/dialog/ResetPasswordAlertDialogFragment.java

```diff
@@ -0,0 +1,190 @@
+package org.edx.mobile.view.dialog;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.content.Context;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.widget.TextInputEditText;
+import android.support.design.widget.TextInputLayout;
+import android.support.v7.app.AlertDialog;
+import android.view.View;
+import android.view.accessibility.AccessibilityEvent;
+
+import com.google.inject.Inject;
+
+import org.edx.mobile.R;
+import org.edx.mobile.authentication.LoginService;
+import org.edx.mobile.http.CallTrigger;
+import org.edx.mobile.http.ErrorHandlingCallback;
+import org.edx.mobile.interfaces.OnActivityResultListener;
+import org.edx.mobile.model.api.ResetPasswordResponse;
+import org.edx.mobile.util.InputValidationUtil;
+import org.edx.mobile.util.NetworkUtil;
+import org.edx.mobile.util.images.ErrorUtils;
+
+public class ResetPasswordAlertDialogFragment extends AlertDialogFragment {
```

Comment:
I don't see _any_ current or future benefits of an abstract `DialogFragment`. The extra keys are further bloat that's forcing us to make the title and message dynamically defined in the arguments, when they're actually static and constant in this implementation.